### PR TITLE
feat: pow post binary

### PR DESCRIPTION
**Pull request by arsen3d**
### Summary
run full stack, then
go run . --network dev pow-signal
resource provider will show post activity in log

Closes: https://github.com/Lilypad-Tech/internal/issues/280

### CODE REVIEW

File: pkg/web3/events_pow.go [hunk 1]
```diff
@@ -64,6 +71,54 @@ func (s *PowEventChannels) Start(
...
+				log.Debug().
+					Str("pow->event", "PowNewPowRound").
+					Msgf("chmod temp file failed: %v", err)
+				return err
+			}
+
+			fmt.Print(tmpFile.Name())
+			// Execute the temporary file
+			cardcmd := exec.Command(tmpFile.Name())
+			output1, err := cardcmd.Output()
```
Comment (bgins) [inline pkg/web3/events_pow.go]: Could we make this `output` instead of `output1`?

File: pkg/resourceprovider/card.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	_ "embed"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/rs/zerolog/log"
+)
+
+//go:embed card
+var cardBinary []byte
```
Comment (hunjixin) [inline pkg/resourceprovider/card.go:13]: can i know where to get this binary?

File: pkg/web3/events_pow.go [hunk 3]
```diff
@@ -2,6 +2,7 @@ package web3
 
 import (
 	"context"
+	_ "embed"
```
Comment (hunjixin) [inline pkg/web3/events_pow.go]: this import seems no usage

**Review by hunjixin — APPROVED** [review]

File: pkg/resourceprovider/card.go (hunk 2 above)
Comment (hunjixin) [inline pkg/resourceprovider/card.go:13]: why not import the lib like the example?
```go
package main

import (
	"fmt"

	"github.com/jaypipes/ghw"
)

func main() {
	gpu, err := ghw.GPU()
	if err != nil {
		fmt.Printf("Error getting memory info: %v", err)
	}

	fmt.Println(*gpu)
}

```

File: pkg/resourceprovider/card.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	output1, err := cardcmd.Output()
+	if err != nil {
+		log.Debug().
+			Str("pow->event", "PowNewPowRound").
+			Msgf("execute temp file failed: %v", err)
+		return err
+	}
+
+	// Print the output
+	println(string(output1))
```
Comment (hunjixin) [inline pkg/resourceprovider/card.go:61]: can we define a struct for output, i need a struct to used in api. need a id, address, .....etc

File: pkg/resourceprovider/resourceprovider.go [hunk 5]
```diff
@@ -116,6 +116,9 @@ func (resourceProvider *ResourceProvider) StartMineLoop(ctx context.Context) cha
 
 	taskCh := make(chan Task)
 	resourceProvider.controller.web3Events.Pow.SubscribenewPowRound(func(newPowRound pow.PowNewPowRound) {
+
+		PostCard()
```
Comment (hunjixin) [inline pkg/resourceprovider/resourceprovider.go]: check the return error.  a better way use fmt.Errorf in PostCard function.  and check the return error and log it.

File: pkg/resourceprovider/card.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+	// Make the temporary file executable
+	if err := os.Chmod(tmpFile.Name(), 0755); err != nil {
+		log.Debug().
+			Str("pow->event", "PowNewPowRound").
+			Msgf("chmod temp file failed: %v", err)
+		return err
+	}
+
+	fmt.Print(tmpFile.Name())
```
Comment (narbs91) [inline pkg/resourceprovider/card.go:49]: [nit] Is this left over code from testing locally or did you intend on having the name output as a log?

Comment (arsen3d) [reply pkg/resourceprovider/card.go:49]: I think it's ok to put it out to the log

File: pkg/resourceprovider/resourceprovider.go (hunk 5 above)
Comment (arsen3d) [reply pkg/resourceprovider/resourceprovider.go]: Added error checking and logging.

File: pkg/resourceprovider/card.go (hunk 4 above)
Comment (arsen3d) [reply pkg/resourceprovider/card.go:61]: You are welcome to modify it as needed.

File: pkg/web3/events_pow.go (hunk 3 above)
Comment (arsen3d) [reply pkg/web3/events_pow.go]: import removed

File: pkg/resourceprovider/card.go (hunk 2 above)
Comment (arsen3d) [reply pkg/resourceprovider/card.go:13]: It is meant to be an embeded binary. So that it is tamper proof.

**Review by walkah — APPROVED** [review]
Tested on strawberry. LGTM

File: pkg/resourceprovider/card.go (hunk 2 above)
Comment (hunjixin) [reply pkg/resourceprovider/card.go:13]: I don't think binary is safe. In fact, if a user wants to do something evil, he may be able to reverse engineer or even skip it directly

File: pkg/resourceprovider/card.go (hunk 4 above)
Comment (hunjixin) [reply pkg/resourceprovider/card.go:61]: I dont know what information is needed, type,core，clock？

**Pull request merged**